tests/vanflow_snooper.py

```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/env python
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+import argparse
+import json
+import logging
+import sys
+from threading import Thread
+
+from proton.handlers import MessagingHandler
+from proton.reactor import Container
+from proton import Message
+
+
+# These mappings come directly from the vanflow.h source. They will need to be
+# updated as new records/attributes/etc are added
+#
+
+record_types = {
+    0: "SITE",
+    1: "ROUTER",
+    2: "LINK",
+    3: "CONTROLLER",
+    4: "LISTENER",
+    5: "CONNECTOR",
+    6: "FLOW",
+    7: "PROCESS",
+    8: "IMAGE",
+    9: "INGRESS",
+    10: "EGRESS",
+    11: "COLLECTOR",
+    12: "PROCESS_GROUP",
+    13: "HOST",
+    14: "LOG",
+}
+
+RECORD_TYPE = 0
+IDENTITY = 1
+
+attribute_types = {
+    RECORD_TYPE: "RECORD_TYPE",
+    IDENTITY: "IDENTITY",
+    2: "PARENT",
+    3: "START_TIME",
+    4: "END_TIME",
+    5: "COUNTERFLOW",
+    6: "PEER",
+    7: "PROCESS",
+    8: "SIBLING_ORDINAL",
+    9: "LOCATION",
+    10: "PROVIDER",
+    11: "PLATFORM",
+    12: "NAMESPACE",
+    13: "MODE",
+    14: "SOURCE_HOST",
+    15: "DESTINATION_HOST",
+    16: "PROTOCOL",
+    17: "SOURCE_PORT",
+    18: "DESTINATION_PORT",
+    19: "VAN_ADDRESS",
+    20: "IMAGE_NAME",
+    21: "IMAGE_VERSION",
+    22: "HOST_NAME",
+    23: "OCTETS",
+    24: "LATENCY",
+    25: "TRANSIT_LATENCY",
+    26: "BACKLOG",
+    27: "METHOD",
+    28: "RESULT",
+    29: "REASON",
+    30: "NAME",
+    31: "TRACE",
+    32: "BUILD_VERSION",
+    33: "LINK_COST",
+    34: "DIRECTION",
+    35: "OCTET_RATE",
+    36: "OCTETS_OUT",
+    37: "OCTETS_UNACKED",
+    38: "WINDOW_CLOSURES",
+    39: "WINDOW_SIZE",
+    40: "FLOW_COUNT_L4",
+    41: "FLOW_COUNT_L7",
+    42: "FLOW_RATE_L4",
+    43: "FLOW_RATE_L7",
+    44: "DURATION",
+    45: "IMAGE",
+    46: "GROUP",
+    47: "STREAM_ID",
+    48: "LOG_SEVERITY",
+    49: "LOG_TEXT",
+    50: "SOURCE_FILE",
+    51: "SOURCE_LINE",
+}
+
+log_severity = {
+    0: "DEBUG",
+    1: "INFO",
+    2: "WARNING",
+    3: "ERROR",
+    4: "CRITICAL"
+}
```

Comment:
Simiilarly here: Consider using an Enum for these as well.

Reply:
Done.